.github/workflows/deploy_to_staging.yml

```diff
@@ -1,45 +1,128 @@
 name: Deploy to Staging
 
 on:
-  push:
-    branches:
-      - master
-    paths-ignore:
-      - 'documentation/**'
-
+  workflow_dispatch:
```

Comment:
Why have we changed this file?

Reply:
> Just wondering why there seems to be more changes than I expected. Other than that it looks good.

@amazimbe the recent rails issue highlighted some issues: because we were rebuilding each time, we couldn't even deploy old versions to staging. This way, we build once and promote though, which is  best practice anyway